app/bundles/CampaignBundle/Controller/CampaignMetricsController.php

```diff
@@ -104,4 +109,23 @@ public function emailHoursAction(
             ]
         );
     }
+
+    public function eventDetailsAction(
+        EventDispatcherInterface $eventDispatcher,
+        EventModel $eventModel,
+        int $objectId,
+    ): JsonResponse {
+        $event = $eventModel->getEntity($objectId);
+
+        if (!$event) {
+            return $this->json([
+                'message' => $this->translator->trans('mautic.core.error.notfound', [], 'flashes'),
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        $eventDetailsAction = new EventPreview($event);
+        $eventDispatcher->dispatch($eventDetailsAction, CampaignEvents::ON_EVENT_PREVIEW_REQUEST);
```

Comment:
Perhaps we could use just the event as the [newer Symfony versions allow](https://symfony.com/doc/6.4/components/event_dispatcher.html#dispatch-the-event)?
```suggestion
        $eventDetailsAction = new EventPreview($event);
        $eventDispatcher->dispatch($eventDetailsAction);
```

In the subscriber you then subscribe to the event FQCN, `EventPreview::class` instead of the artificial event name. What do you think? It makes everything a little simpler and less boilerplaity.

Reply:
Good to know, thanks!